### PR TITLE
Update pydantic code to fix warnings

### PR DESCRIPTION
**Pull request by msabramo**
Before (`pydantic==2.7.1`, `pydantic_core==2.18.2`):

```shell
$ env -i PATH=$PATH poetry run pytest litellm/tests/test_proxy_server.py
...
========================== 10 passed, 2 skipped, 48 warnings in 9.83s ===========================
```

After (`pydantic==2.7.1`, `pydantic_core==2.18.2`):

```shell
$ env -i PATH=$PATH poetry run pytest litellm/tests/test_proxy_server.py
...
=========================== 10 passed, 2 skipped, 1 warning in 9.89s ============================
```

After (`pydantic==1.10.15`, `pydantic_core` not installed):

```
$ env -i PATH=$PATH poetry run pytest litellm/tests/test_proxy_server.py
...
=========================== 10 passed, 2 skipped, 1 warning in 9.13s ============================
```

Note that the 1 remaining warning is fixed by #3602.

### CODE REVIEW

File: litellm/proxy/_types.py [hunk 1]
```diff
@@ -216,7 +217,7 @@ class LiteLLMPromptInjectionParams(LiteLLMBase):
     llm_api_system_prompt: Optional[str] = None
     llm_api_fail_call_string: Optional[str] = None
 
-    @root_validator(pre=True)
+    @model_validator(mode="before")
```
Comment (ghost) [inline litellm/proxy/_types.py:235]: i believe this is pydantic v2 only? @msabramo 

openai sdk supports v1 pydantic - https://github.com/openai/openai-python/blob/906ece0e45235cb170bc062902182eb8b55698b9/pyproject.toml#L12

Any way we can fix the warnings without breaking compatibility

Comment (msabramo) [reply litellm/proxy/_types.py:235]: Hi @krrishdholakia! Ok, take a look at https://github.com/BerriAI/litellm/pull/3611. That fixes just the `root_validator` warnings in pydantic v2, while still working in pydantic v1.

Comment (msabramo) [reply litellm/proxy/_types.py:235]: I went ahead and merged #3611 into this PR, so there's just one PR and I ensured there are no merge conflicts.

File: litellm/tests/test_streaming.py [hunk 2]
```diff
@@ -2214,15 +2214,15 @@ class ToolCalls(BaseModel):
 
 class Delta(BaseModel):
     role: str
-    content: Optional[str]
+    content: Optional[str] = None
```
Comment (ghost) [inline litellm/tests/test_streaming.py]: @msabramo any reason why the pydantic PR also involves setting these values to None? 

We've seen issues during streaming when setting default values like this

Comment (msabramo) [reply litellm/tests/test_streaming.py]: @krrishdholakia Nah, no particular reason. I think those came from me running this:

```shell
bump-pydantic litellm
```

I just went and reverted those changes in bdb995f43.

**Comment by ghost** [comment]
Awesome - is this ready for review? @msabramo

**Comment by msabramo** [comment]
> Awesome - is this ready for review? @msabramo

Yep, go ahead!

**Comment by ghost** [comment]
LGTM! 

Will take care of fixing any ci/cd issues on main

**Comment by ghost** [comment]
Thanks for all your contributions @msabramo

**Pull request merged**